### ISSUE TITLE
[Web] SemanticsBinding.instance.ensureSemantics() is propagating click on overlapping widgets

### ISSUE THREAD

**Issue by rafhaelamarques** [flutter/flutter#163576]
As title says. 

I'm using **Autocomplete** and **ListTile** to exemplify.

### Steps to reproduce

1. Create a app with SemanticsBinding.instance.ensureSemantics() in the main method;
2. Add the Autocomplete and ListTile with onTap();
3. Use Autocomplete and tap an option.

### Expected results

Since the Autocomplete is being used, so the option item should get the user's click.

### Actual results

But what happen is that the click "leaks" to the ListTile under it. The Autocomplete doesn’t identified the click, but the ListTile does.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();

  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          SearchWidget(),
          for (int i = 0; i < 10; i++)
            ListTile(
              onTap: () {
                debugPrint('You just tapped on item $i');
              },
              title: Text('Item $i'),
            ),
        ],
      ),
    );
  }
}

class SearchWidget extends StatelessWidget {
  const SearchWidget({super.key});

  @override
  Widget build(BuildContext context) {
    const List<String> kOptions = <String>[
      'Apple',
      'Banana',
      'Cherry',
      'Date',
      'Fig',
      'Grape',
      'Kiwi',
      'Lemon',
      'Mango',
      'Nectarine',
      'Orange',
      'Peach',
      'Pear',
      'Quince',
      'Raspberry',
      'Strawberry',
      'Tangerine',
      'Ugli fruit',
      'Watermelon',
    ];

    return Autocomplete<String>(
      optionsBuilder: (TextEditingValue textEditingValue) async {
        if (textEditingValue.text.isEmpty) {
          return const Iterable<String>.empty();
        }
        return kOptions.where((String option) {
          return option.contains(textEditingValue.text.toLowerCase());
        });
      },
      onSelected: (item) {
        debugPrint('You just selected $item');
      },
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

Without SemanticsBinding:

https://github.com/user-attachments/assets/5fba7b13-ffd1-4bfb-891d-9eb91c811bdf

With SemanticsBinding:

https://github.com/user-attachments/assets/6ae3945a-3b4d-4c9e-9410-219c2e42cf74

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.29.0, on macOS 15.3.1 24D70 darwin-x64, locale pt-BR) [721ms]
    • Flutter version 3.29.0 on channel stable at /Users/ion/workspace/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 35c388afb5 (8 days ago), 2025-02-10 12:48:41 -0800
    • Engine revision f73bfc4522
    • Dart version 3.7.0
    • DevTools version 2.42.2

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.1) [2,6s]
    • Android SDK at /Users/ion/Library/Android/sdk
    • Platform android-35, build-tools 35.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.4+-12422083-b607.1)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2) [1.521ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [70ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2) [69ms]
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.4+-12422083-b607.1)

[✓] IntelliJ IDEA Community Edition (version 2024.3.3) [66ms]
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.97.2) [30ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.104.0

[✓] Connected device (2 available) [6,8s]
    • macOS (desktop) • macos  • darwin-x64     • macOS 15.3.1 24D70 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 133.0.6943.99

[✓] Network resources [469ms]
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Thanks for filing the issue. Reproduced this on Flutter master version `3.30.0-1.0.pre.191` as well.

**Comment by chunhtai** [comment]
yup this looks like a framework issue

**Comment by yjbanov** [comment]
This has been reported again in https://github.com/flutter/flutter/issues/165447.

**Comment by edeetee** [comment]
Why does ensureSemantics even affect the proportion of normal taps? 
The documentation for ensureSemantics only mentions collecting semantic information. Nothing about changing the behaviour of touch interaction.

I don't think that ensureSemantics should be changing interaction at all. I am calling that method to make semantic information available for screen readers. I don't want any changed interaction behavior.

**Comment by yjbanov** [comment]
@edeetee The screen reader needs to be aware of tappable/clickable nodes, and they need to be able to tap/click on them. So there's actually a DOM `click` listener on the element. What we do is try to disambiguate between DOM clicks and gesture-based clicks. However, when the framework and the semantics tree disagree about the hit test order, the events don't even land on the same widget. So if the semantics tree propagates a click to the wrong widget, it could take effect instead of the hit-test computed by the framework. We should treat this as a severe bug and fix it asap.

**Comment by QuncCccccc** [comment]
Reopen this issue as https://github.com/flutter/flutter/pull/167522 is reverted.

**Comment by Tiska** [comment]
Hi @yjbanov :) we are very interested in this issue at Sncf Connect & Tech, if we can be of any help to tackle the issue don't hesitate to ping :)

**Comment by chunhtai** [comment]
Hi @Tiska, @QuncCccccc  is actively working on this issue. Once they have a pr up, would appreciate if you can give a try to see if it fixes the issue on your side

**Comment by Tiska** [comment]
@chunhtai with great pleasure :)

**Comment by DanhTran0312** [comment]
@QuncCccccc @Tiska Hi, do you have any update on this issue?

**Comment by QuncCccccc** [comment]
Hi, https://github.com/flutter/flutter/pull/173005 is in review. Once google testing is fixed, it's ready to go.

**Comment by Piinks** [comment]
Re-opened as the change was reverted.
